sui-execution/latest/sui-move-natives/src/crypto/ecdsa_k1.rs

```diff
@@ -286,3 +293,74 @@ pub fn secp256k1_verify(
 
     Ok(NativeResult::ok(cost, smallvec![Value::bool(result)]))
 }
+
+pub fn secp256k1_sign(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 3);
+
+    // This function is only used for testing, so we don't need to charge gas
+    let cost = 0.into();
+
+    let hash = pop_arg!(args, u8);
+    let msg = pop_arg!(args, VectorRef);
+    let private_key_bytes = pop_arg!(args, VectorRef);
+
+    let msg_ref = msg.as_bytes_ref();
+    let private_key_bytes_ref = private_key_bytes.as_bytes_ref();
+
+    let sk = match <Secp256k1PrivateKey as ToFromBytes>::from_bytes(&private_key_bytes_ref) {
+        Ok(sk) => sk,
+        Err(_) => return Ok(NativeResult::err(cost, INVALID_PRIVKEY_OR_HASH)),
+    };
+
+    let kp = Secp256k1KeyPair::from(sk);
+
+    let signature = match hash {
+        KECCAK256 => kp.sign_with_hash::<Keccak256>(&msg_ref),
+        SHA256 => kp.sign_with_hash::<Sha256>(&msg_ref),
+        _ => return Ok(NativeResult::err(cost, INVALID_PRIVKEY_OR_HASH)),
+    };
+
+    Ok(NativeResult::ok(
+        cost,
+        smallvec![Value::vector_u8(signature.as_bytes().to_vec())],
+    ))
+}
+
+pub fn secp256k1_keypair_from_seed(
+    _context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    debug_assert!(ty_args.is_empty());
+    debug_assert!(args.len() == 1);
+
+    // This function is only used for testing, so we don't need to charge gas
```

Comment:
is this just for now? I do not understand what it means for a public function to assume it is used only in testing

Reply:
No, this function should never be used for anything but test purposes. The corresponding move functions are annotated with `#[test_only]` but I don't think we can do anything for the Rust functions besides adding a comment.

Reply:
you are right this is a rust function, my bad